### PR TITLE
Create common Connection struct

### PR DESCRIPTION
**Pull request by zgalor**
Initial commit of Connection interface to start working from

cc @yaacov

### CODE REVIEW

**Comment by zgalor** [comment]
@moolitayer @jhernand please review

File: pkg/client/client.go [hunk 1]
```diff
@@ -25,6 +25,8 @@ package client
 // The connection may consume expensive resources, like TCP connections, or file descriptors, so it
 // is important to reuse it as much as possible, and to close it once it is no longer needed.
 type Connection interface {
+	// Connect creates a ne connection to the messaging broker.
+	Connect() error
```
Comment (jhernand) [inline pkg/client/client.go]: How will the caller specify the connection parameters? Can we have something like this:

```go
connection, err := client.NewConnectionBuilder().
  Host("myhost").
  Port(123).
  User("myuser").
  Password("mypass").
  Build()
if err != nil {
  ...
}
```

Comment (zgalor) [reply pkg/client/client.go]: I don't mind though it's very java like. We can also start with a static struct init which is more "GOish"

```
connection := StompConnection{
  brokerHost: "myhost",
  brokerPort: 123,
  user: "myuser",
  password: "mypass",
}

err := connection.Connect()
if err != nil {
  ...
}

**Comment by yaacov** [comment]
Merging, so we can continue working.

We will figure out the best way to init the object on the go. I or @zgalor will open an issue about that.

**Comment by zeari** [comment]
> Merging, so we can continue working.

:clap: :clap: :clap: :clap: :clap: :clap: :clap:

**Pull request merged**